kubemarine/procedures/check_paas.py

```diff
@@ -103,76 +103,65 @@ def _check_same_os(cluster: KubernetesCluster):
-        }
-        containerd_name = "containerd"
-        containerd_name_last = "containerd"
-        if "docker" in cluster.inventory['services']['cri']['containerRuntime']:
-            if version_key == "version_rhel":
-                containerd_name = "containerd.io"
-                containerd_name_last = "containerdio"
-
-            expected_system_packages["docker"] = {
-                "docker": compatibility["docker"][k8s_version][version_key],
-                containerd_name: compatibility[containerd_name_last][k8s_version][version_key]
-                }
-        elif "containerd" in cluster.inventory["services"]["cri"]["containerRuntime"]:
-            if version_key in ["version_rhel", "version_rhel8"]:
-                containerd_name = "containerd.io"
-                containerd_name_last = "containerdio"
-
-            expected_system_packages["containerd"] = {
-                containerd_name: compatibility[containerd_name_last][k8s_version][version_key],
-                "podman": compatibility["podman"][k8s_version][version_key]
-            }
+    version_key = system.get_compatibility_version_key(cluster)
+    if not version_key:
+        raise TestFailure("OS is unknown or multiple OS present")
+    k8s_version = cluster.inventory['services']['kubeadm']['kubernetesVersion']
+    compatibility = cluster.globals["compatibility_map"]["software"]
+    if k8s_version not in compatibility["kubeadm"]:
+        raise TestWarn(f"Using not recommended k8s version: {k8s_version}")
+
+    # Mapping "system_package_alias -> expected_packages_names -> expected_versions"
+    # We assume that system packages have word "haproxy"/"keepalived"/"docker"/"containerd"/"podman" in their name,
+    # if not - then we may miss such package
+    if pckg_alias == "haproxy":
+        expected_system_packages = {"haproxy": compatibility["haproxy"][k8s_version][version_key]}
+    elif pckg_alias == "keepalived":
+        expected_system_packages = {"keepalived": compatibility["keepalived"][k8s_version][version_key]}
+    elif pckg_alias == "containerd":
+        expected_system_packages = {"podman": compatibility["podman"][k8s_version][version_key]}
+        if version_key in ["version_rhel", "version_rhel8"]:
+            expected_system_packages["containerd.io"] = compatibility["containerdio"][k8s_version][version_key]
+        else:
+            expected_system_packages["containerd"] = compatibility["containerd"][k8s_version][version_key]
+    elif pckg_alias == "docker":
+        expected_system_packages = {"docker": compatibility["docker"][k8s_version][version_key]}
+        if version_key in ["version_rhel", "version_rhel8"]:
+            expected_system_packages["containerd.io"] = compatibility["containerdio"][k8s_version][version_key]
+        else:
+            expected_system_packages["containerd"] = compatibility["containerd"][k8s_version][version_key]
```

Comment:
Minor moment: debian & docker is [also](https://github.com/Netcracker/KubeMarine/blob/0.14.0/kubemarine/resources/configurations/defaults.yaml#L348) based on containerd.io

Reply:
fixed